extensions/ql-vscode/src/databases/local-databases-ui.ts

```diff
@@ -150,6 +152,19 @@ class DatabaseTreeDataProvider
         );
       });
 
+      if (
+        this.currentDatabaseItem &&
+        !this.languageContext.isSelectedLanguage(
+          tryGetQueryLanguage(this.currentDatabaseItem.language),
+        )
+      ) {
+        this.currentDatabaseItem = undefined;
+        await this.databaseManager.setCurrentDatabaseItem(
+          this.currentDatabaseItem,
+        );
+        this._onDidChangeTreeData.fire(this.currentDatabaseItem);
+      }
```

Comment:
This seems to work (from local testing), but I'm wondering if the `getChildren` method is the best place for this logic. Should it maybe be handled in the database manager? 😵

Reply:
I don't think the `getChildren` method is the best place for this. Could we listen to the `onLanguageContextChangedEmitter` event in the database manager and deselect it there? That would make more sense to me.

Reply:
Makes sense! I'll have a go at that 🍀